packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2052,7 +2051,9 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         // Note that this should happen before the `this.needsFlush` check below because in the scenario where we are
         // not connected, `this.needsFlush` will be false but the PendingStateManager might have pending messages and
         // hence needs to track this.
-        this.pendingStateManager.onFlush(isImmediateBatch);
+        if (this.flushMode !== FlushMode.Immediate || isImmediateBatch) {
```

Comment:
I know this was refactored out of the pending state manager but can we update the comment here to clarify the 'why'? 

It currently looks very confusing if you're reading the comment about `isImmediateBatch`, implying that this OR statement is redundant, as it reads 'do this if either it's not an immediate batch (flush being called at the end of an batch `FlushMode.Immediate`) or if it's not an immediate batch as flushmode is not set to immediate'.

I guess the question is why does the pending state manager need to know about the flush only when we're either in turnbased flushmode or we're at the end of an ordersequentially batch.

Reply:
I agree it's a little wonky. But I think it needs this protection until [`flush` is no longer a public method](https://dev.azure.com/fluidframework/internal/_workitems/edit/1076). As it stands, anyone can call `flush` on the container runtime regardless of a batch being performed or not.

Once that change happens, we can remove this condition and all the stuff around `isImmediateBatch` since we can ensure `flush` is only called during a batch.